src/jaxsim/rbda/contacts/rigid.py

```diff
@@ -372,64 +333,74 @@ def compute_contact_forces(
                 )
             )
 
+        # Compute the free linear acceleration of the collidable points.
+        # Since we use doubly-mixed jacobian, this corresponds to W_p̈_C.
         free_contact_acc = RigidContacts._linear_acceleration_of_collidable_points(
             BW_nu=BW_ν,
             BW_nu_dot=BW_ν̇_free,
             CW_J_WC_BW=J_WC,
-            CW_J_dot_WC_BW=J̇_WC_BW,
+            CW_J_dot_WC_BW=J̇_WC,
         ).flatten()
 
-        # Compute stabilization term
-        ḣ = velocity[:, 2].squeeze()
+        # Compute stabilization term.
         baumgarte_term = RigidContacts._compute_baumgarte_stabilization_term(
-            inactive_collidable_points=inactive_collidable_points,
-            h=h,
-            ḣ=ḣ,
-            K=self.parameters.K,
-            D=self.parameters.D,
+            inactive_collidable_points=(δ <= 0),
+            δ=δ,
+            δ̇=δ̇,
```

Comment:
See https://github.com/ami-iit/jaxsim/pull/260/files#r1795057708, this is even more confusing:

~~~
δ̇=δ̇,
~~~

Reply:
Same as https://github.com/ami-iit/jaxsim/pull/260#discussion_r1795722266.